### ISSUE TITLE
Roadmap

### ISSUE THREAD

**Issue by Lewiscowles1986** [cleverwise/cya#2]
Is there a roadmap for where this is going?

Are there advantages to this over say a local repo you can install and re-download packages from for the binary repo-available files that are backed up?

**Comment by cleverwise** [comment]
Well at this point it is really just see what people want and the level of interest.

However there isn't much value in repos at this point.  For starters this is just a BASH script and nothing binary (short of all code is binary based ;-))  Plus it doesn't matter your distro or *nix as all can pull from Github.  You can even just do a Github clone/pull and rsync or something.

Did you have some thoughts?

**Comment by Lewiscowles1986** [comment]
I think there was some confusion about my question. 

I meant what are the advantages of backing up system libs and executables using cya over re-installing system from local mirror or online.

To give an example of an existing process; on Debian-based systems `dpkg -l` gets a list of installed packages (using apt). You can also `dpkg --get-selections >~/cya/package-log-debian` and later `dpkg --set-selections <~/cya/package-log-debian` to restore. (I'm sure there is a centos/rpm system equivalent).

I know snap and other systems (including docker) are a bit more of a pain, but containers don't seem to be targeted, and this says it's specifically for system-backups (but I think it looks like it handles snaps). 

I was trying to understand if there is a reason this fits some people better than those solutions, where the holes are in other solutions and where this fits in. (There must be some gap between imaging entire file-systems and desired-state)

Don't get me wrong I'm not down on cya, it looks great. I'd just love to understand why, and where it's designed for, perhaps where it's going, and I feel a roadmap might help communicate that without monopolising on your time.

**Comment by cleverwise** [comment]
CYA has a lot of uses and really can go everywhere from desktops, to servers, to IoT.  For example I have deployed it on Raspberry Pi's, along with desktops and servers.

However to answer your question about using an online or local repo vs CYA.

1) Online Repos - Well here CYA will be faster.  There should be no downloading as the files are stored on the node itself or at least on local network storage in the case of say a SAN mounted block device.

2) Local Repos - Obviously speed won't be as great but most likely the repo isn't stored on the node itself.  With CYA there is zero network bottleneck this is true even if say / and /home are local network mounted blocks.  This is because the copying of files should be done on that device like SAN to SAN.

However regardless repos don't make it so easy to create a working snapshot.  CYA can and will restore you to the exact state as the snapshot.  So let's say you installed Ubuntu 16.04, have been keeping up with the updates, have installed some PPAs, a few snaps, even custom compiled a few apps.

When you use CYA on October 28, 2017 you create an exact snapshot of that very moment.  Now you restore say on October 29, 2017 your restored system will be as it was on October 28, 2017 with all the updates and patches applied up to that time including snaps, PPAs, custom compiled apps, etc.

If you restore from the Ubuntu repo you won't have all your PPAs, custom compiled apps, snaps, themes, etc.  Oh sure you may generate a file of some repo stuff installed but what if you have a custom compiled app? Nope.  What about a custom theme? Maybe or maybe not depending.  What about the state of your log files?  Those will go back to the repo state where as CYA will restore your logs just as they were when the snapshot was taken.

Plus even if that generated list of installed apps is used the reinstall must go down the list installing each one then rolling to the next one.  CYA is just a straight copying the files back.  That is MUCH faster.  In fact in five to ten minutes you can completely return your system back to the same state.  In our tests five minutes was more common.  What repo method can do that with all your custom packages, updates, software?  Now what if you custom edited /etc/fstab to mount a sshfs share.  Is a repo going to do that?  Nope, but CYA will.

So bottom line is CYA is faster period as it is straight copying files.  There is no processing anything.  CYA also takes a complete snapshot of the current the state while using repos will not.
